build/webpack.config.dev.js

```diff
@@ -66,7 +66,8 @@ module.exports = (env) => {
         filename: 'css/app.css'
       }),
       new webpack.DefinePlugin({ 'process.env.APP_URL': JSON.stringify(env.APP_URL) }),
-      new webpack.DefinePlugin({ 'process.env.FIREROAD_URL': JSON.stringify(env.FIREROAD_URL) })
+      new webpack.DefinePlugin({ 'process.env.FIREROAD_URL': JSON.stringify(env.FIREROAD_URL) }),
+      new webpack.DefinePlugin({ 'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV) })
```

Comment:
What's this for?

Reply:
Ah, I needed a way to pass in the current build environment to properly initialize Vuex. Essentially, if this is built for `development`, then it'll enable `strict` mode, while if this is built for `production`, it'll disable it.

This `strict` mode is unrelated to `ECMAScript`'s `strict` mode. `Vuex` `strict` mode means that it throws errors anytime we modify a `store` value without properly calling either a `mutation` or an `action`. This allows us to better debug the code because all of our modifications to these values can be traced to just those `functions` rather than to some random place where we decided to assign the value manually.

We need to disable `strict` mode in `production` since while it's fine for `development`, it adds a lot of `watchers` that weighs down the speed of the application in `production`.